src/constructs/ecs/ecs-task.test.ts

```diff
@@ -75,4 +77,18 @@ describe("The GuEcsTask pattern", () => {
     template.hasGuTaggedResource("AWS::ECS::TaskDefinition", { appIdentity: { app: "ecs-test" } });
     template.hasGuTaggedResource("AWS::ECS::TaskDefinition", { appIdentity: { app: "ecs-test2" } });
   });
+
+  it("should support overriding the subnets used by the task", () => {
+    const stack = simpleGuStackForTesting();
+
+    const vpc2 = Vpc.fromVpcAttributes(stack, "VPC2", {
+      vpcId: "test",
+      availabilityZones: [""],
+      publicSubnetIds: [""],
+      privateSubnetIds: ["abc-123", "def-456"],
+    });
+    generateComplexStack(stack, "ecs-test", makeVpc(stack), vpc2.privateSubnets);
```

Comment:
It looks like we've references to two VPCs here: the return of `makeVpc` and `vpc2`. Is this typical usage? If not, should this test reflect real usage more closely?

Reply:
this was me being lazy and creating a whole vpc in order to get a list if ISubnet I could use to test overriding the subnets. In https://github.com/guardian/cdk/pull/2477/commits/1cfb5ac77a9c83b436fb333390746be78497e42f I do it properly, just creating some subnets to override rather than a whole new vpc